### PR TITLE
fix: update get and delete task unit test and e2e test.

### PR DESCRIPTION
**Pull request by IRONICBo**
## Description

This PR focuses on improving the unit tests for the `manager/job/task.go` and rewriting the e2e tests. The key changes are:
Unit Test Rewrite (`manager/job/task.go`):
1. The unit tests for CreateGetTask and CreateDeleteTask have been completely rewritten.
2. The new tests provide more comprehensive coverage, including handling both success and failure cases.
3. Mocking and assertions have been added to verify internal logic, ensuring the robustness of task creation and deletion workflows.
End-to-End (e2e) Tests:
1. New e2e tests have been added, following the structure of existing preheat tests.
2. The e2e tests now include proper assertions on return values to validate task creation and deletion scenarios.
3. These tests aim to cover the complete flow from request to response, ensuring that the APIs function as expected in real-world scenarios.

## Related Issue

- https://github.com/dragonflyoss/Dragonfly2/issues/2818

## Motivation and Context

The previous unit and e2e tests were either lacking sufficient coverage or not thoroughly validating the expected outcomes. This PR enhances the testing suite by:

Refactoring the unit tests to cover more edge cases.
Introducing e2e tests that assert the correctness of return values and ensure the stability of critical functionalities.
These improvements will help catch bugs earlier and ensure that task creation and deletion APIs behave correctly in both isolated unit tests and integrated environments.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Review by gaius-qi — CHANGES_REQUESTED** [review]
Add failed case.

File: manager/job/task_test.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+func TestTask_CreateGetTask(t *testing.T) {
+	tk := newTask(&job.Job{Server: &machinery.Server{}})
+
+	tests := []struct {
+		name       string
+		schedulers []models.Scheduler
+		args       types.GetTaskArgs
+		expect     *job.GroupJobState
+		wantErr    error
```
Comment (gaius-qi) [inline manager/job/task_test.go:40]: Remove `wantErr`.

File: manager/job/task_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+			},
+			args: types.DeleteTaskArgs{
+				TaskID: "valid-task-id",
+			},
+			expect:  nil,
+			wantErr: errors.New("Result backend required"),
+		},
+	}
+
+	for _, tt := range tests {
```
Comment (gaius-qi) [inline manager/job/task_test.go:127]: tc

File: test/e2e/v2/manager/task.go [hunk 3]
```diff
@@ -126,12 +127,233 @@ var _ = Describe("Get and delete task with Manager", func() {
 			done = waitForDone(job, managerPod)
 			Expect(done).Should(BeTrue())
 
-			// Check delete task response is not null.
-			Expect(job.Result).NotTo(BeNil())
+			// Check delete task response is valid.
+			isValid = checkTaskResponse(job)
+			Expect(isValid).To(BeTrue())
```
Comment (gaius-qi) [inline test/e2e/v2/manager/task.go:132]: Expect(validateTaskResponse()).To(BeTrue())

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: manager/job/task_test.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+	"github.com/RichardKnop/machinery/v1"
+	"github.com/stretchr/testify/assert"
+
+	"d7y.io/dragonfly/v2/internal/job"
+	"d7y.io/dragonfly/v2/manager/models"
+	"d7y.io/dragonfly/v2/manager/types"
+)
+
+func TestTask_CreateGetTask(t *testing.T) {
+	tk := newTask(&job.Job{Server: &machinery.Server{}})
```
[the remark this answers is not in the document]
Comment (IRONICBo) [reply manager/job/task_test.go:33]: OK, updated.

File: manager/job/task_test.go (hunk 1 above)
Comment (IRONICBo) [reply manager/job/task_test.go:40]: OK, updated.

File: test/e2e/v2/manager/task.go (hunk 3 above)
Comment (IRONICBo) [reply test/e2e/v2/manager/task.go:132]: OK, updated.
